parser/src/powdr.lalrpop

```diff
@@ -629,25 +647,42 @@ MatchArms: Vec<MatchArm> = {
 }
 
 MatchArm: MatchArm = {
-    <pattern:Pattern> "=>" <value: Expression> => MatchArm{pattern, value},
+    <pattern:Pattern> "=>" <value: Expression<TermWithoutStruct>> => MatchArm{pattern, value},
 }
 
 IfExpression: Box<Expression> = {
-    <start:@L> "if" <condition:BoxedExpression>
+    <start:@L> "if" <condition:BoxedExpression<TermWithoutStruct>>
         <body:BlockExpression>
         "else"
         <else_body:BlockExpression>
         <end:@R> => ctx.to_expr_with_source_ref(IfExpression{condition, body, else_body}, start, end),
 }
 
-
 BlockExpression: Box<Expression> = {
-    <start:@L> "{" <statements:StatementInsideBlock*> <expr:BoxedExpression?> "}" <end:@R> => ctx.to_expr_with_source_ref(BlockExpression{statements, expr}, start, end),
+    <start:@L> "{" <statements:StatementInsideBlock *> <expr:BoxedExpression<TermWithStruct>?> "}" <end:@R> => ctx.to_expr_with_source_ref(BlockExpression{statements, expr}, start, end),
```

Comment:
Why does it not work to allow structs here?

Reply:
It does. `<TermWithStruct>` means that Structs are possible.